app/src/boardgames/views.py

```diff
@@ -0,0 +1,167 @@
+from django.views.decorators.http import require_http_methods
+from django.contrib.auth.decorators import login_required
+from django.contrib.admin.views.decorators import staff_member_required
+from django.shortcuts import get_object_or_404, redirect, render, reverse
+from .models import Boardgame, Vote
+from .forms import BoardgameForm
+from conferences.models import Zosia
+from users.models import UserPreferences
+from django.contrib import messages
+from django.utils.translation import ugettext_lazy as _
+from django.utils.html import escape
+
+from django.http import Http404, HttpResponse, JsonResponse
+import json
+
+from urllib.request import urlopen
+
+
+@login_required
+@require_http_methods(['GET'])
+def index(request):
+    boardgames = Boardgame.objects.all()
+    boardgames = sorted(boardgames, key=lambda x: x.votes_amount, reverse=True)
+    try:
+        current_zosia = Zosia.objects.find_active()
+        preferences = UserPreferences.objects.get(
+            zosia=current_zosia, user=request.user)
+    except (Zosia.DoesNotExist, UserPreferences.DoesNotExist):
+        ctx = {'boardgames': boardgames}
+    else:
+        paid = preferences.payment_accepted
+        ctx = {'boardgames': boardgames,
+               'paid': paid}
+    return render(request, 'boardgames/index.html', ctx)
+
+
+@login_required
+@require_http_methods(['GET', 'POST'])
+def my_boardgames(request):
+    user_boardgames = Boardgame.objects.filter(user=request.user)
+    can_add = False
+    if user_boardgames.count() < 3:
+        can_add = True
+    ctx = {'user_boardgames': user_boardgames,
+           'can_add': can_add}
+    return render(request, 'boardgames/my_boardgames.html', ctx)
+
+
+# TODO: check with regex and check if not already in the database
+def validate_url(url):
+    pass
+
+
+def get_name(url):
+    boargamegeek_html = urlopen(url).read()
+    title_str = '<title>'
+    encoding = "utf-8"
+    title_bytes = bytearray(title_str, encoding)
+    start_index = boargamegeek_html.find(title_bytes)
+    start_index += len(title_str)
+    end_index = boargamegeek_html.find(
+        bytearray(' |', encoding), start_index)
+    name_bytes = boargamegeek_html[start_index: end_index]
+    name_str = name_bytes.decode(encoding)
+    if name_str == "BoardGameGeek":
+        pass
+        # TODO: jakiś error
+    return name_str
+
+
+@login_required
+@require_http_methods(['GET', 'POST'])
+def create(request):
+    user_boardgames = Boardgame.objects.filter(user=request.user)
+    ctx = {'form': BoardgameForm(request.POST or None)}
+
+    if request.method == 'POST':
+        if ctx['form'].is_valid() and user_boardgames.count() < 3:
+            url = ctx['form'].cleaned_data['url']
+            validate_url(url)
+            name = get_name(url)
+            boardgame = Boardgame(name=name, user=request.user, url=url)
+            boardgame.save()
+            return redirect('my_boardgames')
+        else:
+            pass
```

Comment:
You should add an error message here to be shown in toast (see e.g. views for lectures)